tutorial.md

```diff
@@ -0,0 +1,1039 @@
+    delta    = 3/100  -- Percent of screen to increment by when resizing panes
+```
+
+Now we just need to tell xmonad that we want to use this modified
+`layoutHook` instead of the default.  Again, try to reason this out for
+yourself by just looking at the documentation.  Ready?  Here we go:
+
+``` haskell
+  main :: IO ()
+  main = xmonad $ def
+      { modMask    = mod4Mask  -- Rebind Mod to the Super key
+      , layoutHook = myLayout  -- Use custom layouts
+      }
+    `additionalKeysP`
+      [ ("M-S-z"    , spawn "xscreensaver-command -lock")
+      , ("C-<Print>", unGrab *> spawn "scrot -s"        )
+      , ("<Print>"  , spawn "scrot"                     )
+      ]
+```
+
+But we also wanted to add magnification, right?  Luckily for us, there's
+a module for that as well!  It's called [XMonad.Layout.Magnifier].
+Again, take a look at the documentation yourself before reading on—see
+if you can reason out what to do for yourself.  Let's pick the
+`magnifiercz'` modifier from the library; it magnifies a window by a
+given amount, but only if it is a stack window.  Add it to your three
+column layout thusly:
+
+``` haskell
+myLayout = tiled ||| Mirror tiled ||| Full ||| threeCol
+  where
+    threeCol = magnifiercz' 1.3 $ ThreeColMid nmaster delta ratio
+    tiled    = Tall nmaster delta ratio
+    nmaster  = 1      -- Default number of windows in the master pane
+    ratio    = 1/2    -- Default proportion of screen occupied by master pane
+    delta    = 3/100  -- Percent of screen to increment by when resizing panes
+```
+
+Don't forget to import the module!
+
+You can think of the `$` here as putting everything into parentheses
+from the dollar to the end of the line.  If you don't like that you can
+also write
+
+``` haskell
+  threeCol = magnifiercz' 1.3 (ThreeColMid nmaster delta ratio)
+```
+
+instead.
+
+That's it!  Now we have a perfectly functioning three column layout with
+a magnified stack.  If you compare this with the starting screenshots,
+you will see that that's exactly the behaviour we wanted!
+
+A last thing that that's worth knowing about are so-called
+"combinators"—at least we call them that, we can't tell you what to do.
+These are function that compose with the `xmonad` function and add a lot
+of hooks and other things for you (trying to achieve a specific goal),
+so you don't have to do all the manual work yourself.  For example,
+xmonad—by default—is only [ICCCM] compliant.  Nowadays, however, a lot
+of programs (including many compositors) expect the window manager to
+_also_ be EWMH compliant.  So let's save ourselves a lot of future
+trouble and add that to xmonad straight away!
+
+This functionality is to be found in the [XMonad.Hooks.EwmhDesktops]
+module, so let's import it:
+
+``` haskell
+  import XMonad.Hooks.EwmhDesktops
+```
+
+We might also consider using the `ewmhFullscreen` combinator.  By
+default, a "fullscreened" application is still bound by it's window
```

Comment:
Sorry to be annoying with grammar, but should be "bound by its window"

Reply:
Nothing to worry about; thank you for pointing these out!